File: elucidate-common-lib/src/main/java/com/digirati/elucidate/common/infrastructure/util/PaginationUtils.java
package com.digirati.elucidate.common.infrastructure.util;

import static java.lang.Math.floor;
import static java.lang.Math.max;

public class PaginationUtils {
    public static int calculateLastPage(int totalItems, int pageSize) {
        return (int) floor(max(0, ((double) totalItems - 1)) / pageSize);
    }
}

File: elucidate-common-lib/src/main/kotlin/com/digirati/elucidate/common/infrastructure/util/PaginationUtils.kt
package com.digirati.elucidate.common.infrastructure.util

import kotlin.math.floor

object PaginationUtils {
    @JvmStatic
    fun calculateLastPage(totalItems: Int, pageSize: Int): Int =
        floor(0.0.coerceAtLeast(totalItems.toDouble() - 1) / pageSize).toInt()
}